lib/ui/widgets/donation_countdown_timer.dart

```diff
@@ -37,7 +37,7 @@ class _DonationCountdownTimerState extends State<DonationCountdownTimer>
 
   @override
   void dispose() {
-    _stopTicker();
+    _ticker?.dispose();
```

Comment:
Oh shoot, am I assuming it right that in the previous state, it called itself recursively? 😮 (dispose -> stop -> dispose -> ...)

Reply:
No, not really.

Inside `_stopTicker()`, the `dispose` method is invoked on the `Ticker` instance, releasing its allocated resources. This overridden `dispose` method is triggered when the widget exits the UI tree. Therefore, here we should just free up the `Ticker` resources without cancelling current delivery.